client/rest/test/plugins/metadata/metal_spec.js

```diff
@@ -23,309 +23,255 @@ const {
 			// Arrange:
 			const metadata = getMetadata(70);
-			const fitstKey = metadata.scopedMetadataKey;
+			const firstKey = metadata.scopedMetadataKey;
 			// Act:
-			const { payload } = metal.decode(fitstKey, chunks);
+			const { payload } = metal.decode(firstKey, testData.chunks);
 			// Assert:
-			expect(payload).to.deep.equal(imageBytes);
+			expect(payload).to.deep.equal(testData.imageBytes);
 		});
-		it('can not decodes binary data chunk is broken', () => {
+
+		it('cannot decodes binary data chunk is broken', () => {
 			// Arrange:
 			const metadata = getMetadata(10);
-			const fitstKey = metadata.scopedMetadataKey;
+			const firstKey = metadata.scopedMetadataKey;
 			// delete chunk
 			const deleteChunk = getMetadata(20);
-			const deletedChunks = chunks.filter(obj => JSON.stringify(obj.key) !== JSON.stringify(deleteChunk.scopedMetadataKey));
-			// Act:
-			const result = () => metal.decode(fitstKey, deletedChunks);
-			// Assert:
-			expect(result).to.throw(`Error: The chunk ${deleteChunk.scopedMetadataKey} is missing`);
+			const deletedChunks = testData.chunks.filter(obj => JSON.stringify(obj.key) !== JSON.stringify(deleteChunk.scopedMetadataKey));
+			// Act + Assert:
+			expect(() => metal.decode(firstKey, deletedChunks)).to.throw(`the chunk ${deleteChunk.scopedMetadataKey} is missing`);
 		});
-		it('can not decodes binary data value is broken', () => {
+
+		it('cannot decodes binary data value is broken', () => {
 			// Arrange:
 			const metadata = getMetadata(10);
-			const fitstKey = metadata.scopedMetadataKey;
+			const firstKey = metadata.scopedMetadataKey;
 			// delete chunk value
-			const deleteChunk = chunks.find(obj => 20 === obj.id);
-			deleteChunk.value = deleteChunk.value.slice(0, deleteChunk.value.length - 1);
-			const checksum = metal.generateMetadataKey(new Uint8Array(deleteChunk.value));
-			// Act:
-			const result = () => metal.decode(fitstKey, chunks);
-			// Assert:
-			expect(result).to.throw(`Error: The chunk ${deleteChunk.key} is broken (calculated=${checksum})`);
-		});
-		it('decodes mosaic metadata from chunks', () => {
-			// Arrange:
-			const metadata = getMetadata(130, true);
-			const fitstKey = metadata.scopedMetadataKey;
-			// Act:
-			const { payload } = metal.decode(fitstKey, moasicChunks);
-			// Assert:
-			expect(payload).to.deep.equal(imageBytes);
+			const mutatedChunk = testData.chunks.find(obj => 20 === obj.id);
+			mutatedChunk.value = mutatedChunk.value.slice(0, mutatedChunk.value.length - 1);
+			const checksum = metal.generateMetadataKey(new Uint8Array(mutatedChunk.value));
+			// Act + Assert:
+			expect(() => metal.decode(firstKey, testData.chunks))
+				.to.throw(`the chunk ${mutatedChunk.key} is broken (calculated=${checksum})`);
 		});
 	});
+
 	describe('metal seal', () => {
-		it('metal seal all aggs', () => {
-			// Arrange:
-			const seal = new MetalSeal(1, 'image/png', 'image.png', 'test');
+		const canSealRoundtrip = (seal, hardcodedString) => {
```

Comment:
SealRoundtrip => RoundtripSeal (search and replace)

Reply:
🙏